### DISCUSSION TITLE
Any reference to implement Responder for Vec

### DISCUSSION THREAD

**Discussion by sagarnayak** [rwf2/Rocket#1759] [Questions] [answered]
My user.rs

	use std::io::Cursor;

	use rocket::http::ContentType;
	use rocket::request::Request;
	use rocket::response::{self, Responder, Response};
	use serde::{Deserialize, Serialize};

	#[derive(Serialize, Deserialize, Debug)]
	pub struct User {
		pub id: i64,
		pub name: String,
		pub email: String,
	}

	pub const TABLE_NAME_USER: &str = "users";

	impl<'r> Responder<'r, 'static> for User {
		fn respond_to(self, _: &'r Request<'_>) -> response::Result<'static> {
			let json_string = rocket::serde::json::serde_json::to_string(&self).unwrap();
			Response::build()
				.sized_body(json_string.len(), Cursor::new(json_string))
				.header(ContentType::JSON)
				.ok()
		}
	}

rocket endpoint -

	#[get("/getBlankArray")]
	pub fn get_blank_array() -> status::Custom<content::Json<Vec<User>>> {
		status::Custom(Status::Ok, content::Json(vec![]))
	}

I want to return a vec of users.

Do I need to impl Responder for vec ? or i am on working path?

**Comment by SergioBenitez** [comment, marked as answer]
You never need to use serde_JSON directly. All of the docs guide you to the right path: return `Json<Vec<T>>`, where Json is `rocket::serde::json::Json`.

**Comment by sagarnayak** [comment]
This is what I changed -

	#[get("/getBlankArray")]
	pub fn get_blank_array() -> status::Custom<rocket::serde::json::Json<Vec<User>>> {
		status::Custom(Status::Ok, Json(vec![User { id: 123, name: "test user".to_string(), email: "test@testing.com".to_string() }]))
	}

I got confused here - [Rocket Wrapping](https://rocket.rs/v0.5-rc/guide/responses/#wrapping).
Using `rocket::serde::json::Json` fixed the problem.
thanks @SergioBenitez

**Reply by SergioBenitez** [reply]
You don't need `Custom`.

**Reply by sagarnayak** [reply]
I don't get you. I want to send a custom response code with my response. should i just remove the `Custom`. and write -> `status<rocket::serde::json::Json<Vec<User>>>` ?
